insights/parsers/dmsetup.py

```diff
@@ -121,3 +130,155 @@ def __getitem__(self, idx):
+            if len(device_info_spl) < 3:
+                continue
+            target_type = device_info_spl[2]
+            target_args = device_info_spl[3] if len(device_info_spl) == 4 else None
+            parsed_args = None
+            if target_args:
+                parsed_args = self._parse_target_args(target_type, target_args)
+            self.data.append(SetupStatus(
+                device_name=device_name,
+                start_sector=device_info_spl[0],
+                num_sectors=device_info_spl[1],
+                target_type=target_type,
+                target_args=target_args,
+                parsed_args=parsed_args,
+            ))
+        self.names = [dm[0] for dm in self.data]
+        self.by_name = dict((dm[0], dm) for dm in self.data)
+
+    def _parse_target_args(self, target_type, target_args):
+        pars_func_name = '_parse_target_args_' + target_type.replace('-', '_')
+        pars_func = getattr(self, pars_func_name, None)
+        return pars_func(target_args) if pars_func else None
+
+    def _parse_target_args_thin_pool(self, target_args):
+        """
+        Format:
+            <transaction_id> <used_metadata_blocks>/<total_metadata_blocks>
+            <used_data_blocks>/<total_data_blocks> <held_metadata_root>
+            ro|rw|out_of_data_space [no_]discard_passdown [error|queue]_if_no_space
+            needs_check|- metadata_low_watermark
+        Refer to https://www.kernel.org/doc/Documentation/device-mapper/thin-provisioning.txt .
+        """
+        args = target_args.split()
+        if len(args) < 8 or '/' not in args[1] or '/' not in args[2]:
+            return
+        parsed_args = {}
+        parsed_args['transaction_id'] = args[0]
+        parsed_args['used_metadata_blocks'], parsed_args['total_metadata_blocks'] = args[1].split('/', 1)
+        parsed_args['used_data_blocks'], parsed_args['total_data_blocks'] = args[2].split('/', 1)
+        parsed_args['held_metadata_root'] = args[3]
+        parsed_args['opts'] = args[4:8]
+        parsed_args['metadata_low_watermark'] = args[8] if len(args) > 8 else None
+        return parsed_args
+
+    def _parse_target_args_thin(self, target_args):
+        """
+        Format:
+            <nr_mapped_sectors> <highest_mapped_sector>
+        Refer to https://www.kernel.org/doc/Documentation/device-mapper/thin-provisioning.txt .
+        """
+        args = target_args.split()
+        if len(args) < 2:
+            return
+        parsed_args = {}
+        parsed_args['nr_mapped_sectors'] = args[0]
+        parsed_args['highest_mapped_sector'] = args[1]
+        return parsed_args
+
+    def _parse_target_args_snapshot(self, target_args):
+        """
+        Format:
+            <sectors_allocated>/<total_sectors> <metadata_sectors>
+        Refer to https://www.kernel.org/doc/Documentation/device-mapper/snapshot.txt .
+        """
+        args = target_args.split()
+        if len(args) < 2 or '/' not in args[0]:
```

Comment:
The same as above: here just skip it w/o any prompt info?

Reply:
Changed to capture such lines into `self.unparseable_lines`.